redash/models/__init__.py

```diff
@@ -1168,6 +1172,15 @@ def favorites(cls, user, base_query=None):
     def get_by_slug_and_org(cls, slug, org):
         return cls.query.filter(cls.slug == slug, cls.org == org).one()
 
+    @classmethod
+    def get_by_id_or_slug_and_org(cls, id_or_slug, org):
+        return (
+            cls.query.filter(
+                cast(cls.id, db.String) == id_or_slug, cls.org == org
+            ).one_or_none()
+            or cls.query.filter(cls.slug == id_or_slug, cls.org == org).one()
```

Comment:
This is likely to mess up the index performance (because of the casting to string).

If we redirect from the old /dashboard/ URL to the new /dashboards/ ones, it means we only need to support a GET request by slug and we know when it's just a slug vs. id (let's add a flag to the GET request to note when it's a slug, maybe `?legacy`?). And we can convert all the other API calls to use IDs.

Wdyt?